actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -134,7 +134,7 @@ module CacheHelper
       #
       #   <%= render @notifications, cache: false %>
       def cache(name = {}, options = nil, &block)
-        if controller.perform_caching
+        if controller.try(:perform_caching)
```

Comment:
We avoid to use `try` inside the framework and since we are expecting `controller` to always be defined can you change to a `respond_to?` check?